src/components/List.js

```diff
@@ -26,12 +28,66 @@ const List = ({ token }) => {
   };
 
   const handleCheckbox = async (event) => {
-    const currentDate = Date.now(); // changed to using Date.now(), time given in milliseconds
+    const currentDate = Date.now(); // time given in milliseconds
     const queryCollection = await db.collection(token).doc(event.target.id);
+    // listItem is our current snapshot of the doc
+    const listItem = docs.filter((doc) => doc.id === queryCollection.id)[0];
+    const previouslyPurchasedDate = listItem.lastPurchased;
+
+    /* latestInterval is the number of full days between the two dates
+     * or the timeFrame if the function returns NaN
+     */
+    const latestInterval =
+      differenceInDays(currentDate, previouslyPurchasedDate) ||
+      listItem.timeFrame;
+
+    // If item has not been purchased set to 1, else increment by 1
+    const numberOfPurchases = (listItem.numberOfPurchases || 0) + 1;
+    const daysUntilNextPurchase = calculateEstimate(
+      listItem.timeFrame,
+      latestInterval,
+      numberOfPurchases,
+    );
+
+    // FOR THE DEMO
+    // ------------------------------------------------------------------------------------
+    // update day of demo
+    // const oneWeekAgo = getTime(new Date(2021, 0, 26, 11, 30, 30));
+    // oneWeekAgo simulates the last date of purchase for the item
+    // console.log("Date a week ago: ", toDate(new Date(oneWeekAgo)));
+    // const demoLatestInterval = differenceInDays(currentDate, oneWeekAgo);
+    // const exampleNumberOfPurchases = 7;
+    // const demoEstimate = calculateEstimate(
+    //   listItem.timeFrame,
+    //   demoLatestInterval,
+    //   exampleNumberOfPurchases,
+    // );
+    // demo estimate tells how long from purchase date until we're expected to buy it again
+    // console.log('Demo estimate :', addMilliseconds(previouslyPurchasedDate, demoEstimate));
+    // ------------------------------------------------------------------------------------
+    // END OF DEMO
+
     queryCollection.update({
       lastPurchased: currentDate,
+      timeFrame: daysUntilNextPurchase,
+      numberOfPurchases,
     });
   };
+
+  /* Function updates daysUntilNextPurchase when user reloads the app
+   * Used below when rendering List
+   */
+  const getDaysUntilNextPurchase = (doc) => {
+    const nextPurchasedDate = addDays(doc.lastPurchased, doc.timeFrame);
+
+    /* Added 1 here since we're working with full days
+     * if user taps a checkbox and the timeFrame was 7 days, 6 would have been displayed
+     */
+    const daysUntilNextPurchase =
+      differenceInDays(nextPurchasedDate, Date.now()) + 1;
```

Comment:
This is the documentation on the [differenceInDays](https://date-fns.org/v2.17.0/docs/differenceInDays) function from date-fns